### PR TITLE
Add SpawnedThread

### PR DESCRIPTION
**Pull request by sauljabin**
Spawn threads for each element in a json_array variable

### CODE REVIEW

File: sdk-python/littlehorse/workflow.py [hunk 1]
```diff
@@ -504,6 +510,46 @@ def __init__(self, workflow: "Workflow", initializer: "ThreadInitializer") -> No
...
+            variables={
+                key: to_variable_assignment(value) for key, value in input.items()
+            },
+        )
+
+        node_name = self.add_node(thread_name, start_thread_node)
+        thread_number = self.add_variable(node_name, VariableType.INT)
+        self.mutate(thread_number, VariableMutationType.ASSIGN, NodeOutput(node_name))
+
+        return SpawnedThread(thread_name, thread_number)
```
Comment (coltmcnealy-lh) [inline sdk-python/littlehorse/workflow.py:551]: Nice, wonder if we're going to make this work with Edu's new pattern?

**Review by coltmcnealy-lh — APPROVED** [review]

File: sdk-python/littlehorse/workflow.py (hunk 1 above)
Comment (sauljabin) [reply sdk-python/littlehorse/workflow.py:551]: well, now this PR belongs to Edu 😄

Comment (sauljabin) [reply sdk-python/littlehorse/workflow.py:551]: @eduwercamacaro

File: sdk-python/examples/wait_for_threads/example_wait_for_threads.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    config = LHConfig()
+    config_path = Path.home().joinpath(".config", "littlehorse.config")
+    if config_path.exists():
+        config.load(config_path)
+    return config
+
+
+def get_workflow() -> Workflow:
+    def is_user_group(user_group_thread: ThreadBuilder) -> None:
+        pass
```
Comment (eduwercamacaro) [inline sdk-python/examples/wait_for_threads/example_wait_for_threads.py:27]: WIP

File: sdk-python/examples/parallel_approvals/parallel_approvals.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        thread.execute(TASK_NAME, "var")
+
+    return Workflow("parallel-approvals", my_entrypoint)
+
+
+async def task(arg1: str) -> None:
+    print("This is an example var " + arg1)
+
+
+async def daily_bugle(news: str) -> None:
```
Comment (sauljabin) [inline sdk-python/examples/parallel_approvals/parallel_approvals.py:95]: clean

File: sdk-python/examples/parallel_approvals/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+lhctl run parallel-approvals approvals '[{"userId": "eduwer", "userGroup":"finance"},{"userId": null, "userGroup":"it"}]'
```
Comment (sauljabin) [inline sdk-python/examples/parallel_approvals/README.md:1]: add description

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -50,14 +50,20 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install Dependencies
+        working-directory: ./sdk-python
```
Comment (sauljabin) [inline .github/workflows/main.yml:53]: how did you merge master? it is weird it is showing you changes on master

File: sdk-python/littlehorse/workflow.py [hunk 6]
```diff
@@ -493,6 +511,55 @@ def __str__(self) -> str:
...
+        fixed_threads: Optional[list[SpawnedThread]] = None,
+    ) -> None:
+        self._iterable = iterable
+        self._fixed_threads = fixed_threads
+
+    @classmethod
+    def from_list(cls, *spawned_threads: SpawnedThread) -> "SpawnedThreads":
+        return SpawnedThreads(iterable=None, fixed_threads=list(spawned_threads))
+
+    def build_node(self) -> WaitForThreadsNode:
```
Comment (sauljabin) [inline sdk-python/littlehorse/workflow.py:533]: WaitForThreadsNode is a proto? can you change this method name to compile()?

Comment (eduwercamacaro) [reply sdk-python/littlehorse/workflow.py]: Good suggestion. thanks.

**Pull request merged**